### ISSUE TITLE
Spin out joins as elasticsearch plugin

### ISSUE THREAD

**Issue by ddorian** [crate/crate#7566]
**Use case**:
To be honest, I would consider using crate.io just for the joins, but there is a lot of overhead (keeping _source, docs enabled everywhere, slq, missing es functionality etc).
**Feature description**:
Separate the joins into an elasticsearch plugin.

Makes sense ?

**Comment by mfussenegger** [comment]
This is not really feasible without a huge time investment for which we've little incentive.
We've patched ES in various places and a lot of our functionality assumes that the "ES" parts are in a certain way.

We are further moving in the direction of adapting more ES components to our requirements. Partly to address some of the overhead you mentioned, but also to enable us to build some other features.
It's one of the reasons we've deprecated the possibility to enable the ES API in master already.

Btw., doc-values (column store) can be disabled.
We'd be curious to know which ES functionality you're missing.
